demo_viz.py

```diff
@@ -0,0 +1,196 @@
+"""
+This Python file contains code to generate the 2 visualisations on the page
+https://datacommons.morrisanimalfoundation.org/demographic-data
+titled:
+1. Age Distribution as of <Month> <year>
+2. Sex Status, Baseline vs Year <number>
+as well as the data to generate the graphics titled:
+1. Geographic Distribution of Enrolled Dogs
+2. Urban v/s Suburban v/s Rural
+
+The datasets that power the visualisations and graphics on this page are:
+1. dog_profile.csv
+2. reproductive_history_female.csv
+3. reproductive_history_female.csv
+
+Author: Neha Bhomia
+Created Date: September 19th, 2023
+"""
+
+from py_secrets import dirpath, vizpath, fontpath
+import pandas as pd
+from datetime import date, datetime
+import matplotlib.pyplot as plt
+import seaborn as sns
+import matplotlib.font_manager as font_manager
+import matplotlib.ticker as mtick
+
+# suppressing warnings for category datatype
+# TODO: Investigate this further to prevent technical debt
+import warnings
+warnings.filterwarnings("ignore", "is_categorical_dtype")
+warnings.filterwarnings("ignore", "use_inf_as_na")
+
+# reading in our datasets and force setting data types
+# note: parsing dates adds 01 as default day in date values if date is missing (df_profile raw data date format yyyy-mm)
+df_profile = pd.read_csv(dirpath + "dog_profile.csv", dtype={'subject_id': 'object', 'sex_status': 'category'},
+                         parse_dates=['birth_date', 'enrolled_date'])
+
+df_fem = pd.read_csv(dirpath + "reproductive_history_female.csv", usecols=['subject_id', 'spayed_on_date'],
+                     parse_dates=['spayed_on_date'])
+
+df_male = pd.read_csv(dirpath + "reproductive_history_male.csv", usecols=['subject_id', 'neutered_on_date'],
+                      parse_dates=['neutered_on_date'])
+
+df_tuple = (df_profile, df_fem, df_male)
+
+
+def age_vis(dataframe):
+    """
+    Creates the age distribution visualisation - a bar graph - with age buckets (in years) on the x-axis and
+    dog counts on the y-axis.
+    :param dataframe: the dataset used to generate this visualisation
+    :return: seaborn figure
+    """
+    # calculating age
+    today = date.today()
+    born = dataframe['birth_date'].dt
+    dataframe['current_age'] = today.year - born.year - (today.month < born.month)
+
+    # Calculate the count of dogs at each age
+    age_counts = dataframe['current_age'].value_counts().sort_index()
+
+    # Create a bar chart using Seaborn
+    sns.set_style('whitegrid', {'grid.color': '#ECECEC'})  # HEX code for the grey colour of the grid lines
+    plt.figure(figsize=(8, 6))  # Adjust the figure size as needed
+    sns.barplot(x=age_counts.index, y=age_counts.values, color='#FF5F1F', width=0.7)  # Adjust bar colour using HEX code
```

Comment:
I got `TypeError: bar() got multiple values for argument 'width'` error when I executed this script I think it's because on this line the width is 0.7 and on line 166 the width is set to 0.8. Deleting both width parameters fixed this error. 

Because on line 65 the figure size is defined, it should control the width of the bars.